src/client/sync/mod.rs

```diff
@@ -160,31 +193,43 @@ impl Reader for Context {
 }
 
 impl Writer for Context {
-    fn write_single_register(&mut self, addr: Address, data: Word) -> Result<()> {
+    fn write_single_register(
+        &mut self,
+        addr: Address,
+        data: Word,
+    ) -> io::Result<crate::Result<()>> {
         block_on_with_timeout(
             &self.runtime,
             self.timeout,
             self.async_ctx.write_single_register(addr, data),
         )
     }
 
-    fn write_multiple_registers(&mut self, addr: Address, data: &[Word]) -> Result<()> {
+    fn write_multiple_registers(
+        &mut self,
+        addr: Address,
+        data: &[Word],
+    ) -> io::Result<crate::Result<()>> {
```

Comment:
I wonder if it would make sense to introduce a type alias for `io::Result<crate::Result<T>>`.

Rename `crate::Result` into `ModbusResult` and use `pub type Result<T> = io::Result<ModbusResult<T>>` as nested, crate-level `Result` type?

Reply:
We could add 2 type aliases in `lib.rs`:
- `pub type ModbusResult<T> = Result<T, Exception>;`
- `pub type Result<T> = io::Result<ModbusResult<T>>;`

This will reduce the boilerplate of having to write `io::Result<crate::Result<()>>` everytime.

I'll update it to see if it's easier to understand this way, if not, we can revert the commit.

Reply:
I think we even don't need `ModbusResult` at all.

Because it's only used on the `server` side and it's a simple `Result<Response, Exception>`.

I think we can only keep the type alias `Result<T> = io::Result<result::Result<T, Exception>>`

Reply:
Minimizing the publicly visible and accessible parts of an API is recommended.

The `crate Result` alias is an exception to this rule, because it appears so often in the public (client) API. The client API should be the main focus.